src/SmoFacade/User.cs

```diff
@@ -1,18 +1,62 @@
 ﻿namespace AgDatabaseMove.SmoFacade
 {
+  using System.Collections.Generic;
+  using Smo = Microsoft.SqlServer.Management.Smo;
+
+  public class UserProperties
+  {
+    public string Name { get; set; }
+    public IEnumerable<RoleProperties> Roles { get; set; }
+    public Smo.DatabasePermissionSet Permissions { get; set; }
+    public string LoginName { get; set; }
+  }
+
   public class User
   {
     private readonly Server _server;
-    private readonly Microsoft.SqlServer.Management.Smo.User _user;
+    private readonly Database _database;
+    private Smo.User _user;
 
-    public User(Microsoft.SqlServer.Management.Smo.User user, Server server)
+    public User(Smo.User user, Server server, Database database)
     {
       _user = user;
       _server = server;
+      _database = database;
+    }
+
+    public User(UserProperties userProperties, Server server, Database database)
+    {
+      _server = server;
+      _database = database;
+      ConstructUser(userProperties);
     }
 

```

Comment:
I see what you're doing here...  I have a suggestion, which may or may not be better, but here's what I'm thinking:

Leave the `User` class as a POD (plain old data type).  The User creation code can live in a private/internal method on the `Database` class, where this is currently being called from.

Mainly, I think it's too easy to misuse the class constructor here unintentionally creating a new user

Reply:
I'm inclined to agree. I was modeling after the login creation here, but I think constructor logic should be mostly confined to actions intended to be performed on every instantiation of that object.